### PR TITLE
SQN-1397

### PR DESCRIPTION
**Pull request by lamcc21**
Ticket: [SQN-1397](https://onfinality.atlassian.net/jira/software/projects/SQN/boards/14?assignee=614cf62d071141006ae08a3a&selectedIssue=SQN-1397)

TODO:

- [x] update contract sdk and event parameters once avaliable

### CODE REVIEW

**Comment by mzxyz** [comment]
https://github.com/subquery/network-contracts/pull/192 @lamcc21

**Comment by mzxyz** [comment]
Event changed:

```
// 1. new param `uint256 fee`
event UnbondWithdrawn(address indexed source, uint256 amount, uint256 fee, uint256 index);
// 2. amount changed
event UnbondRequested(address indexed source, address indexed indexer, uint256 amount, uint256 index, UnbondType _type);
// 3. new param `commission`,  `rewards` change to total rewards
event DistributeRewards(address indexed indexer, uint256 indexed eraIdx, uint256 rewards, uint256 commission);
```

File: src/mappings/staking.ts [hunk 1]
```diff
@@ -173,12 +179,28 @@ export async function handleWithdrawRequested(
...
   const id = getWithdrawlId(source, index);
 
+  let updatedAmount = amount;
+
+  if (getWithdrawalType(_type) === WithdrawalType.MERGE) {
+    const staking = Staking__factory.connect(STAKING_ADDRESS, api);
+    const length = (await staking.unbondingLength(indexer)).toNumber();
+
+    // Iterate through unbonding amounts and sum them up
+    for (let i = 0; i < length; i++) {
```
Comment (mzxyz) [inline src/mappings/staking.ts]: This need to be updated

Comment (lamcc21) [reply src/mappings/staking.ts]: updated

**Review by mzxyz — APPROVED** [review]

**Pull request merged**